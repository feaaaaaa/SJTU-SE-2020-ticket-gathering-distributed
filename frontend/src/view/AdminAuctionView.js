import React from 'react';
import {HeaderInfo} from "../component/Header";
import {FooterInfo} from "../component/Footer";
import {AdminAuction} from "../component/AdminAuction";
import {Redirect} from "react-router-dom";
import "../css/Detail.css";
import {RecommendList} from "../component/RecommendList";


export class AdminAuctionView extends React.Component{
    constructor(props) {
        super(props);
        this.state={
            isSearch:false,
            search:null
        };
        this.onSearch=this.onSearch.bind(this);
    }

    onSearch(value){
        this.setState({isSearch:true,search:value});
    }

    render() {
        if(this.state.isSearch){
            console.log("jumping...");
            return <Redirect to={{
                pathname: "/sortPage",
                state:{
                    search:this.state.search
                }
            }}/>;
        }
        return (
            <div>
                <HeaderInfo search={this.onSearch}/>
                <div id="Detail" style={{paddingTop:150,float:"left",marginLeft:-100,width:"80%"}}>
                    <AdminAuction />
                </div>
                <div style={{paddingTop:1000}}>
                    <FooterInfo />
                </div>
            </div>
        );
    }
}