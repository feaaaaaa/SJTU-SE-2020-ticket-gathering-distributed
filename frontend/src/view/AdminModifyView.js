import React from "react";
import {AdminModify} from "../component/AdminModify";
import {HeaderInfo} from "../component/Header";

export class AdminModifyView extends React.Component{
    render() {
        return (
            <div>
                <HeaderInfo/>
                <AdminModify/>
            </div>
        );
    }
}