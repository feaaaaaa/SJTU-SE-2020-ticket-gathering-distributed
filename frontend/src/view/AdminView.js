import React from 'react';
import {Admin} from "../component/Admin";
import {HeaderInfo} from "../component/Header";


export class AdminView extends React.Component{
    render() {
        return (
            <div>
                <HeaderInfo/>
                <div style={{paddingTop:40}}>
                    <Admin/>
                </div>
            </div>
        );
    }
}