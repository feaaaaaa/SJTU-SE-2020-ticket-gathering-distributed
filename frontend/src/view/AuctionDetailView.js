import React from 'react';
import {AuctionDetail} from "../component/AuctionDetail";
import {HeaderInfo} from "../component/Header";


export class AuctionDetailView extends React.Component{
    render(){
        return(
            <div>
                <HeaderInfo/>
                <div style={{paddingTop: 120, width: "80%", margin: "auto"}}>
                    <AuctionDetail/>
                </div>
            </div>
        )
    }
}