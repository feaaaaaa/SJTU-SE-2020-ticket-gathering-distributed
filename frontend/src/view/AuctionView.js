import React from 'react';
import { sports} from "../const/activity";
import {Avatar, Button, Divider, Dropdown, Input, Layout, List, Menu, message} from "antd";
import "../css/sortPage.css"
import "../css/headerInfo.css"
import {RecommendList} from "../component/RecommendList";
import {category_search, search} from "../service/searchService";
import {async} from "fast-glob";
import Redirect from "react-router-dom/es/Redirect";
import { UserOutlined } from '@ant-design/icons';
import {getAuctions} from "../service/AuctionService";
import {SortPageCard} from "../component/SortPageCard";
import {AuctionCard} from "../component/AuctionCard";
import {HeaderInfo} from "../component/Header";
const {Header} = Layout;
const { Search } = Input;


export class AuctionView extends React.Component{
    constructor(props) {
        super(props);
        this.state={
            auctions:[],
            search:null,
            login:false,
            username:null,
            ifauthen:false,
            ifauthor:false,
            isSearch:false,
            interval:2000,
            updateInterval: null
        }
    }

    toggleSearch=(value)=>{
        this.setState({isSearch:true,search:value});
    }

    dateFormat(fmt, date) {
        let ret;
        const opt = {
            "Y+": date.getFullYear().toString(),        // 年
            "m+": (date.getMonth() + 1).toString(),     // 月
            "d+": date.getDate().toString(),            // 日
            "H+": date.getHours().toString(),           // 时
            "M+": date.getMinutes().toString(),         // 分
            "S+": date.getSeconds().toString()          // 秒
            // 有其他格式化字符需求可以继续添加，必须转化成字符串
        };
        for (let k in opt) {
            ret = new RegExp("(" + k + ")").exec(fmt);
            if (ret) {
                fmt = fmt.replace(ret[1], (ret[1].length == 1) ? (opt[k]) : (opt[k].padStart(ret[1].length, "0")))
            };
        };
        return fmt;
    }

    componentDidMount() {
        // this.handleRender();
        const callback = data =>{
            if (data != null) {
                let trueData = data.data;
                console.log("auctionView return:" + JSON.stringify(data));
                let array = [];
                for (let key in trueData) {
                    let obj = trueData[key];
                    let ddl=obj.ddl.substr(0,19).replace('T',' ');
                    let date = new Date();
                    // let test=new Date();
                    // test.setDate(25);
                    // test.setHours(23);
                    // test.setMinutes(0);
                    // test.setSeconds(0);
                    // let fmtDateTest=this.dateFormat("YYYY-mm-dd HH:MM:SS", test);
                    let fmtDateNow=this.dateFormat("YYYY-mm-dd HH:MM:SS", date);
                    // console.log("test:");console.log(fmtDateTest);
                    // console.log("ddl:");console.log(ddl);
                    if(fmtDateNow<ddl)
                        array.push(obj);
                }
                console.log("map to array:");
                console.log(array);
                if (data.status === 200) this.setState({auctions: array});
                else message.error(data.msg);
            }
        }
        getAuctions(localStorage.getItem("token"),callback);
        const token=localStorage.getItem("token");
        let errorCnt=0;
        this.state.updateInterval=setInterval(()=>{
            getAuctions(token,(data)=>{
                if (data != null) {
                    let trueData = data.data;
                    console.log("fresh:" + JSON.stringify(data));
                    let array = [];
                    for (let key in trueData) {
                        let obj = trueData[key];
                        let ddl=obj.ddl.substr(0,19).replace('T',' ');
                        let date = new Date();
                        let fmtDateNow=this.dateFormat("YYYY-mm-dd HH:MM:SS", date);
                        if(fmtDateNow<ddl)
                            array.push(obj);
                    }
                    if (data.status === 200) this.setState({auctions: array});
                    else {
                        message.error(data.msg);
                        errorCnt++;
                        if(errorCnt>3)
                            this.setState({interval:3000});
                        if(errorCnt>10)
                            this.setState({interval:2*60*1000});
                    }
                }
            })
        },this.state.interval);
    }

    componentWillUnmount() {
        clearInterval(this.state.updateInterval);
    }

    // handleRender = () =>{
    //     const callback = data =>{
    //         if (data != null) {
    //             let trueData = data.data;
    //             console.log("auctionView return:" + JSON.stringify(data));
    //             let array = [];
    //             for (let key in trueData) {
    //                 let obj = trueData[key];
    //                 obj.id = key;
    //                 array.push(obj);
    //             }
    //             console.log("map to array:");
    //             console.log(array);
    //             if (data.status === 200) this.setState({auctions: array});
    //             else message.error(data.msg);
    //         }
    //     }
    //     getAuctions(localStorage.getItem("token"),callback);
    // }

    getRenderCallback = (value) =>{
        if(value ===1)
            console.log("render again");
            this.handleRender();
    }

    render() {
        if(this.state.isSearch){
            console.log("jumping...");
            return <Redirect to={{
                pathname: "/sortPage",
                state:{
                    search:this.state.search
                }
            }}/>;
        }
        return(
            <div >
                <HeaderInfo search={value => this.toggleSearch(value)}/>
                <Divider plain className="divider"> </Divider>
                <div id="sortPageDiv" style={{paddingBottom:100,paddingTop:150}}>
                    <List
                        grid={{gutter: 10, column: 1}}
                        pagination={{
                            onChange: page => {
                                console.log(page);
                            },
                            pageSize: 10,
                        }}
                        footer={
                            <div style={{textAlign:"center"}}>
                                <b>O_ligei ! </b> footer part
                            </div>
                        }
                        dataSource={this.state.auctions}
                        renderItem={item => (
                            <List.Item>
                                <AuctionCard info={item} getRenderCallback ={this.getRenderCallback.bind(this)}/>
                            </List.Item>
                        )}
                    />
                </div>
            </div>
        )
    }
}