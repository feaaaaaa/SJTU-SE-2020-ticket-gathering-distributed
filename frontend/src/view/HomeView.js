import React from 'react';
import {HeaderInfo} from "../component/Header";
import {FooterInfo} from "../component/Footer";
import { Carousel,Divider,BackTop } from 'antd';
import "../css/home.css"
import {HomeCategoryPage} from "../component/HomeCategoryPage";
import {Navigate} from "../component/Navigate";

export class HomeView extends React.Component{
    onChange(a,b,c){
        // console.log(a,b,c)
    }

    render(){
        return(
            <div>
                <HeaderInfo/>
                <Divider plain className="divider"> </Divider>
                <div id="carousel">
                    <Carousel autoplay afterChange={this.onChange.bind(this)}>
                        <div id="carouselDiv">
                            <img src={require('../resources/carousel.png')} alt="carousel1" id="carouselImg"/>
                        </div>
                        <div>
                            <img src={require('../resources/carousel2.jpg')} alt="carousel2" id="carouselImg"/>
                        </div>
                    </Carousel>
                </div>
                <Navigate/>
                <div id="homeCategoryPageDiv">
                    <HomeCategoryPage/>
                </div>
                {/*<div style={{paddingTop:1500}}>*/}
                    <FooterInfo/>
                    <BackTop/>
                {/*</div>*/}
            </div>
        )
    }
}
