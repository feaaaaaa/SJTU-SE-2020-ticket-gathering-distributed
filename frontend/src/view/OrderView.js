import React from 'react';

export class OrderView extends React.Component{
    render(){
        return(
            <div>Orders</div>
        )
    }
}