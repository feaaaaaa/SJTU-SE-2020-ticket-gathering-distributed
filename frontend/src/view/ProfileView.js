/**
 *@param this.state.userInfo.balance {number}
 */
import React from 'react';
import {Descriptions, Badge, message,Button,InputNumber, Modal} from "antd";
import "../css/profile.css"
import {HeaderInfo} from "../component/Header";
import {getPersonInfo} from "../service/userService";
import {Redirect} from "react-router-dom";
import {joinAuctions} from "../service/AuctionService";
import {recharge} from "../service/userService";

export class ProfileView extends React.Component{
    constructor(props) {
        super(props);
        this.state={
            increment:0,
            visible:false,
            userInfo:null,
            ifauthen:false,
            logOut:false,
            isSearch:false,
            search:null,
            clickAuction:true,
            balance:0,
        }
        this.onSearch=this.onSearch.bind(this);
        this.logOut=this.logOut.bind(this);
    }

    logOut(){
        this.setState({logOut:true});
    }

    onSearch(value){
        this.setState({isSearch:true,search:value});
    }

    async componentDidMount() {
        const callback = async (data) => {
            if(data.status===-100 ||data.status===201){
                await this.setState({ifauthen: true});
                localStorage.clear();
            } else await this.setState({
                userInfo: data.data,
                balance:data.data.balance
            })
        };
        let userId = localStorage.getItem("userId");
        await getPersonInfo(userId, localStorage.getItem("token"), callback);
    }

    showModal = () => {
        this.setState({
            visible: true,
        });
    };

    handleOk = e => {
        if(this.state.clickAuction){
            this.setState({clickAuction:false});
            const callback = data => {
                if (data != null) {
                    if(data.status!==200)
                         message.error(data.msg);
                    this.setState({visible: false});
                    this.setState({balance:data.data});
                }
            }
            recharge(parseInt(localStorage.getItem("userId")),this.state.increment,localStorage.getItem("token"),callback);
            setTimeout(()=>{this.setState({clickAuction:true})}, 2000);
        }else message.error("点击太快了，休息一会吧")
    };

    handleCancel = e => {
        console.log(e);
        this.setState({
            visible: false,
        });
    };

    onChange = value => {
        this.setState({increment:value});
        console.log('changed', value);
    }


    render() {
        if(this.state.isSearch){
            console.log("jumping...");
            return <Redirect to={{
                pathname: "/sortPage",
                state:{
                    search:this.state.search
                }
            }}/>;
        }
        if(this.state.logOut){
            return <Redirect to={{pathname: "/"}}/>;
        }
        else{
            if(this.state.ifauthen){
                message.error("请先登录");
                return <Redirect to={{pathname: "/login"}}/>;
            }
            else if (this.state.userInfo == null) {
                return (
                    <p>404</p>
                )
            }
            else {
                return (
                    <div>
                        <HeaderInfo logOut={this.logOut} search={this.onSearch}/>
                        <div style={{paddingTop:150}}>
                            <div id="profileDiv">
                                <Descriptions title="个人信息" bordered>
                                    <Descriptions.Item label="用户名">{this.state.userInfo.username}</Descriptions.Item>
                                    <Descriptions.Item label="性别">{this.state.userInfo.gender}</Descriptions.Item>
                                    <Descriptions.Item label="用户类型">{this.state.userInfo.type?("用户"):("管理员")}</Descriptions.Item>
                                    <Descriptions.Item label="电话号码">{this.state.userInfo.phone}</Descriptions.Item>
                                    <Descriptions.Item label="邮箱" span={2}>
                                        {this.state.userInfo.email}
                                    </Descriptions.Item>
                                    {/*<Descriptions.Item label="Status" span={3}>*/}
                                    {/*    <Badge status="processing" text="Running" />*/}
                                    {/*</Descriptions.Item>*/}
                                    <Descriptions.Item label="账户余额">
                                        ${this.state.balance}
                                        <Button onClick={this.showModal}>充值</Button>
                                        <Modal
                                            title="充值"
                                            visible={this.state.visible}
                                            onOk={this.handleOk}
                                            onCancel={this.handleCancel}
                                        >
                                            <InputNumber
                                                defaultValue={this.state.increment}
                                                formatter={value => `$ ${value}`.replace(/\B(?=(\d{3})+(?!\d))/g, ',')}
                                                parser={value => value.replace(/\$\s?|(,*)/g, '')}
                                                onChange={this.onChange}
                                            />
                                        </Modal>
                                    </Descriptions.Item>
                                    {/*<Descriptions.Item label="Discount">$20.00</Descriptions.Item>*/}
                                    {/*<Descriptions.Item label="Official Receipts">$60.00</Descriptions.Item>*/}
                                    <Descriptions.Item label="头像">
                                        <img alt="userIcon" style={{height:100}} src={this.state.userInfo.personIcon}/>
                                    </Descriptions.Item>
                                </Descriptions>
                            </div>
                        </div>
                    </div>
                )
            }
        }
    }
}