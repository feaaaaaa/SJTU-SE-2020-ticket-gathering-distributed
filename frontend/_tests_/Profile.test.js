jest.mock('node-fetch');
import React from 'react'
import {ProfileView} from "../src/view/ProfileView";
import fetch from 'node-fetch';

const {Response} = jest.requireActual('node-fetch');

describe("test profile",() => {
    const wrapper=shallow(<ProfileView/>);
    it("test initialized state",()=>{
        expect(wrapper.state().userInfo).toBeNull();
        expect(wrapper).toMatchSnapshot();
    });

    it("test componentDidMount function",async()=>{
        const json={
            'username':'oligei',
            'gender':'Male',
            'type':'user',
            'phone':'123456',
            'email':'oligei@gmail.com'
        };
        fetch.mockReturnValue(Promise.resolve(new Response(JSON.stringify(json))));
        await wrapper.instance().componentDidMount();
        console.log("test");
        expect(wrapper.state().userInfo.username).toEqual('oligei');
        expect(wrapper.state().userInfo.gender).toEqual('Male');
        expect(wrapper.state().userInfo.type).toEqual('user');
        expect(wrapper.state().userInfo.phone).toEqual('123456');
        expect(wrapper.state().userInfo.email).toEqual('oligei@gmail.com');
        expect(wrapper).toMatchSnapshot();
    });

});