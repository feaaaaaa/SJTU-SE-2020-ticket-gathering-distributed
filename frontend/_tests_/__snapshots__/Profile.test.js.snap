// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test profile test componentDidMount function 1`] = `
<div>
  <HeaderInfo
    logOut={[Function]}
  />
  <div
    style={
      Object {
        "paddingTop": 50,
      }
    }
  >
    <div
      id="profileDiv"
    >
      <Descriptions
        bordered={true}
        title="个人信息"
      >
        <DescriptionsItem
          label="用户名"
        >
          oligei
        </DescriptionsItem>
        <DescriptionsItem
          label="性别"
        >
          Male
        </DescriptionsItem>
        <DescriptionsItem
          label="用户类型"
        >
          用户
        </DescriptionsItem>
        <DescriptionsItem
          label="电话号码"
        >
          123456
        </DescriptionsItem>
        <DescriptionsItem
          label="邮箱"
          span={2}
        >
          oligei@gmail.com
        </DescriptionsItem>
        <DescriptionsItem
          label="账户余额"
        >
          $80.00
        </DescriptionsItem>
        <DescriptionsItem
          label="Config Info"
        >
          <img
            alt="userIcon"
            style={
              Object {
                "height": 100,
              }
            }
          />
        </DescriptionsItem>
      </Descriptions>
    </div>
  </div>
</div>
`;

exports[`test profile test initialized state 1`] = `
<p>
  404
</p>
`;

exports[`test register test componentDidMount function 1`] = `
<div>
  <HeaderInfo />
  <div
    style={
      Object {
        "paddingTop": 50,
      }
    }
  >
    <div
      id="profileDiv"
    >
      <Descriptions
        bordered={true}
        title="个人信息"
      >
        <DescriptionsItem
          label="用户名"
        >
          oligei
        </DescriptionsItem>
        <DescriptionsItem
          label="性别"
        >
          Male
        </DescriptionsItem>
        <DescriptionsItem
          label="用户类型"
        >
          用户
        </DescriptionsItem>
        <DescriptionsItem
          label="电话号码"
        >
          123456
        </DescriptionsItem>
        <DescriptionsItem
          label="邮箱"
          span={2}
        >
          oligei@gmail.com
        </DescriptionsItem>
        <DescriptionsItem
          label="账户余额"
        >
          $80.00
        </DescriptionsItem>
        <DescriptionsItem
          label="Config Info"
        >
          <img
            alt="userIcon"
            style={
              Object {
                "height": 100,
              }
            }
          />
        </DescriptionsItem>
      </Descriptions>
    </div>
  </div>
</div>
`;

exports[`test register test initialized state 1`] = `
<p>
  404
</p>
`;
